### PR TITLE
[IMP] l10n_th: add PromptPay QR code payment for invoice

### PR DESCRIPTION
**Pull request by tong-odoo**
Impacted Version:

- 17.0 and above

This commit improve below features:

- Add PromptPay QR code payment for invoice based on below github repository https://github.com/kittinan/thai-qr-payment

task-3165449

### CODE REVIEW

File: addons/l10n_th/models/res_bank.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+        return super()._get_error_messages_for_qr(qr_method, debtor_partner, currency)
+
+    def _check_for_qr_code_errors(self, qr_method, amount, currency, debtor_partner, free_communication, structured_communication):
+        if qr_method == 'emv_qr' and self.country_code == 'TH' and self.proxy_type not in ['ewallet_id', 'merchant_tax_id', 'mobile']:
+            return _("The PayNow Type must be either Ewallet ID, Merchant Tax ID or Mobile Number to generate a Thailand Bank QR code")
+
+        return super()._check_for_qr_code_errors(qr_method, amount, currency, debtor_partner, free_communication, structured_communication)
+
+# 00020101021229370016A0000006770101110113006699620338253037645802TH6304563C
```
Comment (vin-odoo) [inline addons/l10n_th/models/res_bank.py:63]:
```suggestion
```
You forgot to remove this :smile:

**Review by vin-odoo — COMMENTED** [review]
One small comment, beside that lgtm

**Comment by tong-odoo** [comment]
@vin-odoo Thanks for the review. The comment is updated!

**Review by vin-odoo — APPROVED** [review]
@robodoo r+

**Comment by tong-odoo** [comment]
@robodoo retry

**Pull request closed without merging**